### PR TITLE
[wpilib] update to main

### PR DESCRIPTION
**Pull request by autoantwort**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/wpilib/fix-fmtlib-10.patch [hunk 1]
```diff
@@ -0,0 +1,14 @@
+diff --git a/wpiutil/src/main/native/include/wpi/Synchronization.h b/wpiutil/src/main/native/include/wpi/Synchronization.h
+index 62a79efd9..f3c3bf143 100644
+--- a/wpiutil/src/main/native/include/wpi/Synchronization.h
++++ b/wpiutil/src/main/native/include/wpi/Synchronization.h
+@@ -442,6 +442,9 @@ class SignalObject final {
+   T m_handle;
+ };
+ 
++template<typename T>
++auto format_as(const SignalObject<T> &s) { return s.GetHandle(); }
```
Comment (Adela0814) [inline ports/wpilib/fix-fmtlib-10.patch:10]: I noticed that you have already submitted this changes to upstream, but this PR has been closed. see https://github.com/wpilibsuite/allwpilib/pull/5433.

File: ports/wpilib/portfile.cmake [hunk 2]
```diff
@@ -1,59 +1,46 @@
...
+vcpkg_cmake_config_fixup(PACKAGE_NAME wpimath)
+vcpkg_cmake_config_fixup(PACKAGE_NAME wpinet)
+vcpkg_cmake_config_fixup(PACKAGE_NAME wpiutil)
 
 vcpkg_copy_pdbs()
 
-file(INSTALL "${SOURCE_PATH}/LICENSE.txt" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include" "${CURRENT_PACKAGES_DIR}/debug/share")
+
+file(INSTALL "${SOURCE_PATH}/LICENSE.md" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```
Comment (Adela0814) [inline ports/wpilib/portfile.cmake:46]:
```suggestion
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE.md")
```

**Review by Adela0814 — CHANGES_REQUESTED** [review]

**Comment by Adela0814** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

**Comment by autoantwort** [comment]
Depends on https://github.com/microsoft/vcpkg/pull/33433

**Comment by Adela0814** [comment]
I encountered the following issues while testing usage:
Repro dtrps:
`vcpkg install wpilib[*]:x64-windows`
Usage(provided by vcpkg):
```
    # this is heuristically generated, and may not be correct
    find_package(apriltag CONFIG REQUIRED)
    target_link_libraries(main PRIVATE apriltag)

    find_package(cameraserver CONFIG REQUIRED)
    target_link_libraries(main PRIVATE cameraserver)

    find_package(cscore CONFIG REQUIRED)
    target_link_libraries(main PRIVATE cscore)

    find_package(hal CONFIG REQUIRED)
    target_link_libraries(main PRIVATE hal)

    find_package(ntcore CONFIG REQUIRED)
    target_link_libraries(main PRIVATE ntcore)

    find_package(wpilibNewCommands CONFIG REQUIRED)
    target_link_libraries(main PRIVATE wpilibNewCommands)

    find_package(wpilibc CONFIG REQUIRED)
    target_link_libraries(main PRIVATE wpilibc)

    find_package(wpimath CONFIG REQUIRED)
    target_link_libraries(main PRIVATE wpimath)

    find_package(wpinet CONFIG REQUIRED)
    target_link_libraries(main PRIVATE wpinet)

    find_package(wpiutil CONFIG REQUIRED)
    target_link_libraries(main PRIVATE wpiutil)
```
Error:
```
1> CMake generation started for configuration: 'x64-Debug'.
1> Command line: "C:\WINDOWS\system32\cmd.exe" /c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Ninja"  -DCMAKE_BUILD_TYPE:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="E:\CMakeTest\out\install\x64-Debug" -DCMAKE_TOOLCHAIN_FILE:STRING="E:\\5\vcpkg\\scripts\\buildsystems\\vcpkg.cmake" -DCMAKE_C_COMPILER:FILEPATH="C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/x64/cl.exe" -DCMAKE_CXX_COMPILER:FILEPATH="C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/x64/cl.exe"  -DCMAKE_MAKE_PROGRAM="C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\Ninja\ninja.exe" "E:\CMakeTest" 2>&1"
1> Working directory: E:\CMakeTest\out\build\x64-Debug
1> [CMake] CMake Error at E:/5/vcpkg/scripts/buildsystems/vcpkg.cmake:859 (_find_package):
1> [CMake]   Found package configuration file:
1> [CMake] 
1> [CMake]     E:/5/vcpkg/installed/x64-windows/share/ntcore/ntcore-config.cmake
1> [CMake] 
1> [CMake]   but it set ntcore_FOUND to FALSE so package "ntcore" is considered to be
1> [CMake]   NOT FOUND.  Reason given by package:
1> [CMake] 
1> [CMake]   The following imported targets are referenced, but are missing: wpinet
1> [CMake] 
1> [CMake] Call Stack (most recent call first):
1> [CMake]   C:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.24/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
1> [CMake]   E:/5/vcpkg/installed/x64-windows/share/cameraserver/cameraserver-config.cmake:4 (find_dependency)
1> [CMake]   E:/5/vcpkg/scripts/buildsystems/vcpkg.cmake:859 (_find_package)
1> [CMake]   CMakeTest/CMakeLists.txt:20 (find_package)
1> [CMake] -- Configuring incomplete, errors occurred!
1> [CMake] See also "E:/CMakeTest/out/build/x64-debug/CMakeFiles/CMakeOutput.log".
1> [CMake] See also "E:/CMakeTest/out/build/x64-debug/CMakeFiles/CMakeError.log".
1> 'C:\WINDOWS\system32\cmd.exe' '/c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Ninja"  -DCMAKE_BUILD_TYPE:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="E:\CMakeTest\out\install\x64-Debug" -DCMAKE_TOOLCHAIN_FILE:STRING="E:\\5\vcpkg\\scripts\\buildsystems\\vcpkg.cmake" -DCMAKE_C_COMPILER:FILEPATH="C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/x64/cl.exe" -DCMAKE_CXX_COMPILER:FILEPATH="C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/x64/cl.exe"  -DCMAKE_MAKE_PROGRAM="C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\Ninja\ninja.exe" "E:\CMakeTest" 2>&1"' execution failed with error: ''C:\WINDOWS\system32\cmd.exe' '/c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Ninja"  -DCMAKE_BUILD_TYPE:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="E:\CMakeTest\out\install\x64-Debug" -DCMAKE_TOOLCHAIN_FILE:STRING="E:\\5\vcpkg\\scripts\\buildsystems\\vcpkg.cmake" -DCMAKE_C_COMPILER:FILEPATH="C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/x64/cl.exe" -DCMAKE_CXX_COMPILER:FILEPATH="C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/x64/cl.exe"  -DCMAKE_MAKE_PROGRAM="C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\Ninja\ninja.exe" "E:\CMakeTest" 2>&1"' returned with exit code: 1'.

```

**Comment by Adela0814** [comment]
All feature tested pass in the following triplets:

- x64-windows
- x64-windows-static
- x86-windows

**Comment by autoantwort** [comment]
@Adela0814 Does it work now?

**Comment by Adela0814** [comment]
New Error:
```
1> [CMake] CMake Error at E:/5/vcpkg/installed/x64-windows/share/wpilibNewCommands/wpilibNewCommands-config.cmake:11 (include):
1> [CMake]   include could not find requested file:
1> [CMake] 
1> [CMake]     E:/5/vcpkg/installed/x64-windows/share/wpimath/wpilibNewCommands.cmake
```

**Comment by autoantwort** [comment]
I have now tested it locally, so it should work now

**Comment by Adela0814** [comment]
Usage tested pass on `x64-windows`.

**Review by Adela0814 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**